### ISSUE TITLE
Quarkus 3.6.7 runs all concurrent http request in the same vert.x-event-loop-thread 

### ISSUE THREAD

**Issue by diallama** [quarkusio/quarkus#41486]
### Describe the bug

I am developing a reactive service using Quarkus 3.6.7  and admit each request is extremely fast, during a load test (200 requests in parallel, which in my opinion is negligible), I notice that all requests are executed by the same event-loop-thread despite the fact that other event-loop-threads are available (6 at application startup). This behavior leads to an increase in server response time (up to 7 seconds), while the endpoint doesn't do much. I am using shell script to loop http request without delay to benchmark the service and JProfiler to profile. What I noticed in the profiler thread history view is that for each bechmark run ALL requests  are executed in the same vertx-eventloop-thread. Interestingly each run picks a different event-loop-thread, but all requests are executed in that event-loop-thread.

Additionally:

- Receives incoming requests on RESTEasy-reactive REST APIs.
- My RESTEasy-reactive  APIs are not blocking (Uni return types)

### Expected behavior
No delays.requests are processed in parallel on all available I/O threads.

_No response_

### Quarkus version or git rev

3.6.5

### Additional information

![image](https://github.com/quarkusio/quarkus/assets/34651525/f2ca503e-0841-4bf9-9f10-1bbd8cf05f6a)
![image](https://github.com/quarkusio/quarkus/assets/34651525/08f4e892-2b9b-4feb-a149-58b31f90f6df)
![image](https://github.com/quarkusio/quarkus/assets/34651525/fcf2ef38-d9f2-4f5e-940f-59792e49413c)
![image](https://github.com/quarkusio/quarkus/assets/34651525/54deaafa-8b26-41dd-86d8-541dfd340f01)

**Comment by diallama** [comment]
Additionally:

    - Receives incoming requests on RESTEasy REST APIs.
    - My RESTEasy  APIs are not blocking (Uni return types)

**Comment by gsmet** [comment]
The first thing you need to try is to update to a more recent version of Quarkus.

I would try 3.8.5 as a first step as it's our current LTS and then 3.12.0 which is the latest if you still experience the behavior.

If you still encounter the issue then we will need to investigate what's going on.

**Comment by diallama** [comment]
thanks @gsmet  for your feedback. i'll do the test and keep you posted.

**Comment by diallama** [comment]
i've tested with both versions of quarkus but the problem remains the same. however, when testing with the GrettingRessource project, the problem doesn't seem to arise (use of all available threads). It may be linked to something else on my side (any idea is welcome) because in addition to the asynchronous processing I do on the side, my producer resource does exactly the same thing as the hello below. 

@Path("/hello")
@ApplicationScoped
@NonBlocking
public class GreetingResource {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<Response> hello() {
        final Response response = Response.ok().build();
        return Uni.createFrom().item(response);
    }
}

![image](https://github.com/quarkusio/quarkus/assets/34651525/70eac817-0a24-4e80-8dea-fd76714b1343)

**Comment by geoand** [comment]
`curl` is likely not making requests fast enough. 

Trying using [wrk2](https://github.com/giltene/wrk2) or something similar

**Comment by diallama** [comment]
@geoand  on some runs, the same thread seems to be used very often. however, when I replay the same test, there's sometimes an alternation of available threads at the start. 


![image](https://github.com/quarkusio/quarkus/assets/34651525/4f5ffb03-a5e9-45c4-beaf-82986719a19b)

![image](https://github.com/quarkusio/quarkus/assets/34651525/b671c1b8-c3b9-4f50-b8a3-5210dac2f3f3)

**Comment by geoand** [comment]
>  Wouldn't there be some kind of affinity to activate/deactivate

Affinity WRT what?

**Comment by diallama** [comment]
I've noticed that if the http client multiplexes several http requests on the same TCP connection, these are processed by the same event-loop-thread, which results in long response times. however, if the requests are spread over several TCP connections, then more event-loop-thread comes into play. this suggests that an event-loop-thread manages a TCP connection and not an event in general. 

first test: you will notice that all 86753 requests are handled by the same event-loop-thread

![image](https://github.com/quarkusio/quarkus/assets/34651525/b6113a46-b1f2-44c2-b338-4d69fa9e07c5)
![image](https://github.com/quarkusio/quarkus/assets/34651525/718ca3dc-b1e6-4883-bc21-7c98ae0e403c)

second test: you'll notice that the two available event-loop-threads share the 77822 requests 

![image](https://github.com/quarkusio/quarkus/assets/34651525/5cebc70b-aa74-43dd-9c52-3d398ad83153)
![image](https://github.com/quarkusio/quarkus/assets/34651525/eeb782db-714b-4e78-a094-e23fec5bac38)


NB: I've sorted the response times in descending order

**Comment by cescoffier** [comment]
WRK is keeping the connection open, if I'm not mistaken. So, once a connection is established, it keeps the same event loop. 

Anyway, I just tried with:

```java
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public Uni<String> hello() {
        System.out.println("Thread: " + Thread.currentThread().getName());
        return Uni.createFrom().item("Hello " + Thread.currentThread().getName());
    }
```

With:

```
>  wrk --latency -t 10 -c 100 http://localhost:8080/hello
```

And I get all the event loop used (in prod mode, we use less event loop in dev mode):

```
...
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-9
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-5
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-2
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-4
Thread: vert.x-eventloop-thread-3
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-8
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-7
Thread: vert.x-eventloop-thread-1
Thread: vert.x-eventloop-thread-6
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-0
Thread: vert.x-eventloop-thread-0
...
```

The test was done with Quarkus 3.14.4.

I'm going to close this issue as out of date. If you see this behavior again, please open another issue with a reproducer.
